test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestIntrinsics.java

```diff
@@ -70,30 +70,30 @@ public boolean test1(Class<?> supercls, Class<?> subcls) {
     @DontCompile
     public void test1_verifier(boolean warmup) {
         Asserts.assertTrue(test1(java.util.AbstractList.class, java.util.ArrayList.class), "test1_1 failed");
-        Asserts.assertTrue(test1(MyValue1.ref.class, MyValue1.ref.class), "test1_2 failed");
-        Asserts.assertTrue(test1(MyValue1.class, MyValue1.class), "test1_3 failed");
-        Asserts.assertTrue(test1(MyValue1.ref.class, MyValue1.class), "test1_4 failed");
-        Asserts.assertFalse(test1(MyValue1.class, MyValue1.ref.class), "test1_5 failed");
+        Asserts.assertTrue(test1(MyValue1.class, MyValue1.class), "test1_2 failed");
+        Asserts.assertTrue(test1(MyValue1.class.asValueType(), MyValue1.class.asValueType()), "test1_3 failed");
+        Asserts.assertTrue(test1(MyValue1.class.asValueType(), MyValue1.class.asValueType()), "test1_4 failed");
```

Comment:
should the first MyValue1.class.asValueType() in line 75 be MyValue1.class (to match MyValue1.ref.class) ?

Reply:
MyValue1.class.asValueType() and MyValue1.class are different (secondary mirror vs primary mirror), so changing from MyValue1.class.asValueType() to MyValue1.class would make this line to fail.
However, this line has not been fixed properly (it accidentally duplicates line 74).
Now, I think the right way to fix it is:

Asserts.assertFalse(test1(MyValue1.class, MyValue.class.asValueType()), "test1_4 failed");

The test has to be inverted from assertTrue to assertFalse to verify that the two mirrors are distinct.

Reply:
In fact `MyValue1.class` == `ldc QMyValue1;` and `MyValue1.class` and `MyValue1.class.asValueType()` are the secondary mirror.  `MyValue1.ref.class` is the primary mirror.

AFAIU, these tests fail in -Xcomp mode (pass in -Xint mode) as it's awaiting for the JIT phase 2 support (JDK-8267932) which is blocked by the core reflection support.

Would disabling -Xcomp run from these tests be a better temporay fix until JDK-8267932?